msg/CMakeLists.txt

```diff
@@ -178,12 +175,81 @@ add_custom_command(OUTPUT ${uorb_sources}
 	VERBATIM
 	)
 
-add_library(uorb_msgs STATIC EXCLUDE_FROM_ALL ${uorb_sources})
-add_dependencies(uorb_msgs uorb_headers prebuild_targets)
+px4_add_library(uorb_msgs ${uorb_sources})
+add_dependencies(uorb_msgs uorb_headers)
 
-# microCDR now part of uORB
-target_include_directories(uorb_msgs
-	PRIVATE ${PX4_SOURCE_DIR}/src/lib/micro-CDR/include
-	PRIVATE ${PX4_BINARY_DIR}/src/lib/micro-CDR/include/microcdr
+###############################################################################
+# micro-cdr serialization
+###############################################################################
+# if RTPS is enabled generate serialization code for each uORB message
+if (GENERATE_RTPS_BRIDGE)
+
+	set(uorb_headers_microcdr)
+	set(msg_out_path_microcdr ${PX4_BINARY_DIR}/uORB_microcdr/topics)
+	set(uorb_sources_microcdr)
+	set(msg_source_out_path_microcdr ${CMAKE_CURRENT_BINARY_DIR}/topics_microcdr_sources)
+
+	# send topic files
+	set(send_topic_files)
+	foreach(topic ${config_rtps_send_topics})
+		set(msg_file ${PX4_SOURCE_DIR}/msg/${topic}.msg)
+		list(APPEND send_topic_files ${msg_file})
+		get_filename_component(msg ${msg_file} NAME_WE)
+		list(APPEND uorb_headers_microcdr ${msg_out_path_microcdr}/${msg}.h)
```

Comment:
Isn't `${msg}` just the same as `${topic}`?

Reply:
right